### PR TITLE
Use autotyping to add type hint annotations based on common names

### PR DESCRIPTION
**Pull request by namurphy**
This PR is one in a series that use `autotyping` to add automatically add type hints annotations. This one uses the option to guess type hint annotations from common parameter names. For example, if a parameter ends with `name`, then it usually assumes that it is a string.  

The other PRs were pretty safe, but this one needs a more careful review.

### CODE REVIEW

File: plasmapy/analysis/time_series/running_moments.py [hunk 1]
```diff
@@ -17,7 +17,7 @@
 _run_moment_tuple = namedtuple("Running_Moment", ["run_moment", "time"])
 
 
-def running_mean(signal, radius):
+def running_mean(signal, radius: float):
```
Comment (namurphy) [inline plasmapy/analysis/time_series/running_moments.py:20]:
```suggestion
def running_mean(signal, radius: int):
```

File: plasmapy/analysis/time_series/running_moments.py [hunk 2]
```diff
@@ -62,7 +62,7 @@ def running_mean(signal, radius):
     return run_mean[window - 1 :] / window
 
 
-def running_moment(signal, radius, moment=1, time=None):
+def running_moment(signal, radius: float, moment=1, time=None):
```
Comment (namurphy) [inline plasmapy/analysis/time_series/running_moments.py:65]:
```suggestion
def running_moment(signal, radius: int, moment=1, time=None):
```

File: plasmapy/analysis/time_series/tests/test_running_moments.py [hunk 3]
```diff
@@ -16,14 +16,14 @@
         (np.array([-0.5, -0.0, 0.5, 1.0, 1.5]), 2, [0.5]),
     ],
 )
-def test_running_mean(signal, radius, expected) -> None:
+def test_running_mean(signal, radius: float, expected) -> None:
```
Comment (namurphy) [inline plasmapy/analysis/time_series/tests/test_running_moments.py:19]:
```suggestion
def test_running_mean(signal, radius: int, expected) -> None:
```

File: plasmapy/analysis/time_series/tests/test_running_moments.py [hunk 4]
```diff
@@ -16,14 +16,14 @@
...
     """Test running_mean function"""
     result = running_mean(signal=signal, radius=radius)
     assert np.allclose(result, expected)
 
 
 @pytest.mark.parametrize(("signal", "radius"), [([1, 2], 1), ([1, 2, 3, 4], 1.2)])
-def test_running_mean_exception(signal, radius) -> None:
+def test_running_mean_exception(signal, radius: float) -> None:
```
Comment (namurphy) [inline plasmapy/analysis/time_series/tests/test_running_moments.py:26]:
```suggestion
def test_running_mean_exception(signal, radius: int) -> None:
```

File: plasmapy/analysis/time_series/tests/test_running_moments.py [hunk 5]
```diff
@@ -46,7 +46,7 @@ def test_running_mean_exception(signal, radius) -> None:
         ([1, 2, 3, 2, 1] * u.eV, 1, 4, [1, 2, 3, 4, 5], ([3] * u.eV, [3])),
     ],
 )
-def test_running_moment(signal, radius, moment, time, expected) -> None:
+def test_running_moment(signal, radius: float, moment, time, expected) -> None:
```
Comment (namurphy) [inline plasmapy/analysis/time_series/tests/test_running_moments.py:49]:
```suggestion
def test_running_moment(signal, radius: int, moment, time, expected) -> None:
```

File: plasmapy/analysis/time_series/tests/test_running_moments.py [hunk 6]
```diff
@@ -61,7 +61,7 @@ def test_running_moment(signal, radius, moment, time, expected) -> None:
         ([1, 2, 3, 4, 5], 1, 2, [1, 2, 3, 4]),
     ],
 )
-def test_running_moment_exception(signal, radius, moment, time) -> None:
+def test_running_moment_exception(signal, radius: float, moment, time) -> None:
```
Comment (namurphy) [inline plasmapy/analysis/time_series/tests/test_running_moments.py:64]:
```suggestion
def test_running_moment_exception(signal, radius: int, moment, time) -> None:
```

File: plasmapy/diagnostics/charged_particle_radiography/tests/test_synthetic_radiography.py [hunk 7]
```diff
@@ -203,7 +206,7 @@ def run_mesh_example(
     wire_diameter=20 * u.um,
     mesh_hdir=None,
     mesh_vdir=None,
-    nparticles=1e4,
+    nparticles: int = 1e4,
     problem="electrostatic_gaussian_sphere",
 ):
```
Comment (namurphy) [inline plasmapy/diagnostics/charged_particle_radiography/tests/test_synthetic_radiography.py:211]:
```suggestion
) -> None:
```

File: plasmapy/formulary/braginskii.py
Comment (namurphy) [inline plasmapy/formulary/braginskii.py:1]: `braginskii.py` is going to require some special care, so I should:

 - [ ] Revert this file

File: plasmapy/diagnostics/charged_particle_radiography/tests/test_synthetic_radiography.py [hunk 8]
```diff
@@ -203,9 +206,9 @@ def run_mesh_example(
     wire_diameter=20 * u.um,
     mesh_hdir=None,
     mesh_vdir=None,
-    nparticles=1e4,
+    nparticles: int = 1e4,
```
Comment (namurphy) [inline plasmapy/diagnostics/charged_particle_radiography/tests/test_synthetic_radiography.py:209]:
```suggestion
    nparticles: int = 10000,
```

File: plasmapy/analysis/tests/test_fit_functions.py [hunk 9]
```diff
@@ -44,7 +44,7 @@ def test_is_abc(self) -> None:
             ("root_solve", False),
         ],
     )
-    def test_methods(self, name, isproperty) -> None:
+    def test_methods(self, name: str, isproperty: bool) -> None:
```
Comment (ejohnson-96) [inline plasmapy/analysis/tests/test_fit_functions.py:47]: Is isproperty a boolean?

File: plasmapy/analysis/tests/test_fit_functions.py [hunk 10]
```diff
@@ -121,7 +121,7 @@ def test_repr(self) -> None:
...
         """Test attribute/method/property existence."""
```
Comment (ejohnson-96) [inline plasmapy/analysis/tests/test_fit_functions.py:125]: See above comment

File: plasmapy/analysis/time_series/running_moments.py [hunk 11]
```diff
@@ -17,7 +17,7 @@
 _run_moment_tuple = namedtuple("Running_Moment", ["run_moment", "time"])
 
 
-def running_mean(signal, radius):
+def running_mean(signal, radius: int):
     """
```
Comment (ejohnson-96) [inline plasmapy/analysis/time_series/running_moments.py:21]: Is the radius allowed to be a float?

File: plasmapy/analysis/time_series/tests/test_running_moments.py [hunk 12]
```diff
@@ -61,7 +61,7 @@ def test_running_moment(signal, radius, moment, time, expected) -> None:
         ([1, 2, 3, 4, 5], 1, 2, [1, 2, 3, 4]),
     ],
 )
-def test_running_moment_exception(signal, radius, moment, time) -> None:
+def test_running_moment_exception(signal, radius: int, moment, time) -> None:
     """Test whether exception is risen"""
     with pytest.raises(ValueError):
         running_moment(signal, radius, moment, time)
```
Comment (ejohnson-96) [inline plasmapy/analysis/time_series/tests/test_running_moments.py:67]: Same here, radius is integer

File: plasmapy/formulary/braginskii.py [hunk 13]
```diff
@@ -331,7 +332,7 @@ def __init__(  # noqa: PLR0912, PLR0915
         hall_e=None,
         hall_i=None,
         mu=None,
-        theta=None,
+        theta: Optional[float] = None,
         coulomb_log_method="classical",
```
Comment (ejohnson-96) [inline plasmapy/formulary/braginskii.py:336]: I like how it also specifies if an arguments is optional or not

File: plasmapy/utils/decorators/checks.py [hunk 14]
```diff
@@ -1325,7 +1325,11 @@ def wrapper(*args, **kwargs):
     return decorator(func) if func else decorator
 
 
-def _check_relativistic(V, funcname, betafrac: float = 0.05):
+def _check_relativistic(
+    V: u.Quantity[u.m / u.s],
+    funcname: str,
```
Comment (ejohnson-96) [inline plasmapy/utils/decorators/checks.py:1330]: It also does astropy units? that's pretty cool

**Review by ejohnson-96 — APPROVED** [review]
Flagged a couple I think it did incorrectly, besides that I really like this implementation. Could be very useful for people just starting 👍

File: plasmapy/analysis/time_series/running_moments.py (hunk 11 above)
Comment (namurphy) [reply plasmapy/analysis/time_series/running_moments.py:21]: In this case, `radius` refers to the number of grid points to use in each direction when creating the running mean, so it does need to be an `int`.  This confused me a bit too, so I think this is going to be a place where the type hint is helpful.

File: plasmapy/analysis/tests/test_fit_functions.py (hunk 9 above)
Comment (namurphy) [reply plasmapy/analysis/tests/test_fit_functions.py:47]: Yep!  It is a `bool`.  This is a good one to double check.

**Review by namurphy — COMMENTED** [review]
Thank you for the review!

**Pull request merged**